pkg/plugin/bundle.go

```diff
@@ -31,31 +31,38 @@ type bundle struct {
 	deprecateWarning         string
 }
 
-type BundleOption func(*bundle)
-
-func WithName(name string) BundleOption {
-	return func(opts *bundle) {
-		opts.name = name
+type BundlePlugin struct {
+	Options struct {
+		name                     string
+		version                  Version
+		plugins                  []Plugin
+		supportedProjectVersions []config.Version
+		deprecateWarning         string
 	}
 }
```

Comment:
Wondering if we need the nested struct. Wdyt of making it this?
```suggestion
type BundlePluginOptions struct {
		name                     string
		version                  Version
		plugins                  []Plugin
		supportedProjectVersions []config.Version
		deprecateWarning         string
}
```